modules/aerodyn/src/AeroDyn_Inflow_C_Binding.f90

```diff
@@ -339,42 +318,43 @@ subroutine ADI_C_PreInit(NumTurbines_C, TransposeDCM_in, PointLoadOutput_in, Deb
    ! if (allocated(NacLoadMesh      )) deallocate(NacLoadMesh      )
-   allocate(BldStrMotionMesh(  Sim%NumTurbines), STAT=ErrStat2); if (Failed0('BldStrMotionMesh'  )) return
-   allocate(BldStrLoadMesh(    Sim%NumTurbines), STAT=ErrStat2); if (Failed0('BldStrLoadMesh'    )) return
-   allocate(BldStrLoadMesh_tmp(Sim%NumTurbines), STAT=ErrStat2); if (Failed0('BldStrLoadMesh_tmp')) return
-   ! allocate(NacMotionMesh(    Sim%NumTurbines), STAT=ErrStat2); if (Failed0('NacMotionMesh'    )) return
-   ! allocate(NacLoadMesh(      Sim%NumTurbines), STAT=ErrStat2); if (Failed0('NacLoadMesh'      )) return
+   allocate(BldStrMotionMesh(  Sim%NumTurbines), STAT=ErrStat_F2); if (Failed0('BldStrMotionMesh'  )) return
+   allocate(BldStrLoadMesh(    Sim%NumTurbines), STAT=ErrStat_F2); if (Failed0('BldStrLoadMesh'    )) return
+   allocate(BldStrLoadMesh_tmp(Sim%NumTurbines), STAT=ErrStat_F2); if (Failed0('BldStrLoadMesh_tmp')) return
+   ! allocate(NacMotionMesh(    Sim%NumTurbines), STAT=ErrStat_F2); if (Failed0('NacMotionMesh'    )) return
+   ! allocate(NacLoadMesh(      Sim%NumTurbines), STAT=ErrStat_F2); if (Failed0('NacLoadMesh'      )) return
 
    if (allocated(Map_BldStrMotion_2_AD_Blade   )) deallocate(Map_BldStrMotion_2_AD_Blade  )
    if (allocated(Map_AD_BldLoad_P_2_BldStrLoad )) deallocate(Map_AD_BldLoad_P_2_BldStrLoad)
    ! if (allocated(Map_NacPtMotion_2_AD_Nac    )) deallocate(Map_NacPtMotion_2_AD_Nac    )
-   ! allocate(Map_NacPtMotion_2_AD_Nac(Sim%NumTurbines),STAT=ErrStat2); if (Failed0('Map_AD_BldLoad_P_2_BldStrLoad')) returns
+   ! allocate(Map_NacPtMotion_2_AD_Nac(Sim%NumTurbines),STAT=ErrStat_F2); if (Failed0('Map_AD_BldLoad_P_2_BldStrLoad')) returns
 
    ! Allocate the StrucPtsToBladeMapType array used for mapping structural points to blades of the rotor
    if (allocated(StrucPts_2_Bld_Map)) deallocate(StrucPts_2_Bld_Map)
-   allocate(StrucPts_2_Bld_Map(Sim%NumTurbines), STAT=ErrStat2); if (Failed0('StrucPts_2_Bld_Map'  )) return
+   allocate(StrucPts_2_Bld_Map(Sim%NumTurbines), STAT=ErrStat_F2); if (Failed0('StrucPts_2_Bld_Map'  )) return
 
-   call SetErr(ErrStat,ErrMsg,ErrStat_C,ErrMsg_C)
+   call SetErrStat_F2C(ErrStat_F,ErrMsg_F,ErrStat_C,ErrMsg_C)
 
 contains
    logical function Failed()
-      CALL SetErrStat( ErrStat2, ErrMsg2, ErrStat, ErrMsg, RoutineName )
-      Failed = ErrStat >= AbortErrLev
+      CALL SetErrStat( ErrStat_F2, ErrMsg_F2, ErrStat_F, ErrMsg_F, RoutineName )
+      Failed = ErrStat_F >= AbortErrLev
       if (Failed) then
          call ClearTmpStorage()
-         call SetErr(ErrStat,ErrMsg,ErrStat_C,ErrMsg_C)
+         call SetErrStat_F2C(ErrStat_F,ErrMsg_F,ErrStat_C,ErrMsg_C)
       endif
    end function Failed
 
    ! check for failed where /= 0 is fatal
    logical function Failed0(txt)
+   ! TODO: This does not move the error the C variables; this looks incorrect
       character(*), intent(in) :: txt
-      if (errStat /= 0) then
```

Comment:
@andrew-platt Is it correct that this is checking whether `errStat` is 0? Where it's called above (line 342), the command is setting the error to `ErrStat2`. Since `ErrStat` is initialized to 0, I think this function would only correctly error when NumTurbines is incorrect since that's the only place that `Failed()` is called.

If this is indeed incorrect, I'm happy to rework it, but I just wanted to double check that I'm not misreading it.